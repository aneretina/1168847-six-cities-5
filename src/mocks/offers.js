export const offers = [
  {
    id: 15,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.38333,
        longitude: 4.9,
      }
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
    },
    pictures: [`img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`],
    previewImage: `img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: true,
    price: 180,
    title: `Wood and stone place`,
    type: `room`,
    rating: 3,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    bedroomsCount: 2,
    guestsLimit: 2,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Parking`, `Flowers`, `Dishwasher`, `Towels`, `TV`],
    host: {
      avatar: `img/avatar-max.jpg`,
      name: `Max`,
      isPro: false,
    }
  },

  {
    id: 2,
    city: {
      name: `Paris`,
      location: {
        latitude: 48.864716,
        longitude: 2.349014,
      }
    },
    location: {
      latitude: 48.87456,
      longitude: 2.413,
    },
    pictures: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`],
    previewImage: `img/apartment-02.jpg`,
    isPremium: true,
    isFavorite: true,
    price: 80,
    title: `Beautiful & luxurious apartment in Paris`,
    type: `Apartment`,
    rating: 4,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    bedroomsCount: 1,
    guestsLimit: 2,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Parking`, `Flowers`, `Dishwasher`, `Towels`, `TV`],
    host: {
      avatar: `img/avatar-maggy.jpg`,
      name: `Maggy`,
      isPro: true,
    }
  },

  {
    id: 9,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.935173,
        longitude: 6.953101,
      }
    },
    location: {
      latitude: 50.9682,
      longitude: 6.874,
    },
    pictures: [`img/apartment-03.jpg`, `img/apartment-03.jpg`, `img/apartment-03.jpg`, `img/apartment-03.jpg`, `img/apartment-03.jpg`, `img/apartment-03.jpg`],
    previewImage: `img/apartment-03.jpg`,
    isPremium: false,
    isFavorite: false,
    price: 220,
    title: `Beautiful & luxurious apartment in Cologne`,
    type: `Apartment`,
    rating: 1,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    bedroomsCount: 4,
    guestsLimit: 5,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Parking`, `Flowers`, `Dishwasher`, `Towels`, `TV`],
    host: {
      avatar: `img/avatar-bill.jpg`,
      name: `Bill`,
      isPro: true,
    }
  },
];
