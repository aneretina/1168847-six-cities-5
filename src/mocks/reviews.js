export default [
  {
    id: 1,
    name: `Jake`,
    avatar: [`https://placeimg.com/640/480/people/101`,
      `https://placeimg.com/640/480/people/102`,
      `https://placeimg.com/640/480/people/103`,
      `https://placeimg.com/640/480/people/104`,
    ],
    rating: 3,
    text: `We would like to thank all the staff, who were very friendly during our stay.`,
    date: `10.10.2019`,
  },

  {
    id: 2,
    name: `Mary`,
    avatar: [`https://placeimg.com/640/480/people/101`,
      `https://placeimg.com/640/480/people/102`,
      `https://placeimg.com/640/480/people/103`,
      `https://placeimg.com/640/480/people/104`,
    ],
    rating: 2,
    text: `The hotel had everything we needed and it was in an excellent location, being in a lovely area with lots of places to enjoy ourselves.`,
    date: `10.09.2019`,
  },

  {
    id: 3,
    name: `Oliver`,
    avatar: [`https://placeimg.com/640/480/people/101`,
      `https://placeimg.com/640/480/people/102`,
      `https://placeimg.com/640/480/people/103`,
      `https://placeimg.com/640/480/people/104`,
    ],
    rating: 5,
    text: `Iâ€™ve been many times and I will come again and again.`,
    date: `10.09.2020`,
  },


];
