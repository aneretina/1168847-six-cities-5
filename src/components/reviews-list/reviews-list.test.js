import React from "react";
import renderer from "react-test-renderer";
import {reviews} from "../../mocks/reviews";
import ReviewsList from "./reviews-list";

it(`Should OfferReviewsList render properly`, () => {

  const tree = renderer
    .create(
        <ReviewsList
          reviews={reviews}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
