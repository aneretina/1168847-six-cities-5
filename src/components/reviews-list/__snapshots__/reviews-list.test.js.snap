// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should OfferReviewsList render properly 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      3
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Tom
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We would like to thank all the staff, who were very friendly during our stay.
        </p>
        <time
          className="reviews__time"
          dateTime="2020-10-27"
        >
          October 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Peter
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          The hotel had everything we needed and it was in an excellent location, being in a lovely area with lots of places to enjoy ourselves.
        </p>
        <time
          className="reviews__time"
          dateTime="2020-10-27"
        >
          October 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Michael
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          I’ve been many times and I will come again and again.
        </p>
        <time
          className="reviews__time"
          dateTime="2020-10-27"
        >
          October 2020
        </time>
      </div>
    </li>
  </ul>,
]
`;
